packages/material-ui/src/Button/Button.js

```diff
@@ -330,6 +334,7 @@ const Button = React.forwardRef(function Button(inProps, ref) {
       disabled={disabled}
       focusRipple={!disableFocusRipple}
       focusVisibleClassName={clsx(classes.focusVisible, focusVisibleClassName)}
```

Comment:
> [@espipj] Do we need focus-visible styles when inclusiveDisabled?

Inclusive disabled buttons still allow focus with the keyboard, therefore it needs a focus indicator. Otherwise, people might not know what's the currently focused element.

This prop can be used in more scenarios besides tooltips elements. That means we can't assume that there will  always exist a visual clue from a surrounding element (eg the tooltip). For that reason, yes, add the focus indicator, even if it's redundant in some cases. _Redundant is better than none._ Does this make sense to you?

Reply:
Yep, that's exactly the argument I would make. Though I'm always grappling with the concept of "cursor" and how it relates to mouse and keyboard inputs. We remove the hover styles for mouse cursor as well which is to me the same as focus styles for the "keyboard cursor". 

But I am still leaning towards having different visuals for hover and focus styles because the mouse cursor is inherently visible. The keyboard cursor has no inherent visibility so it makes sense to always display them. Removing focus-visible styles is like hiding the mouse cursor once you hover over an element.